lib/bolt/config.rb

```diff
@@ -147,6 +147,13 @@ def validate
       unless %w[human json].include? self[:format]
         raise Bolt::CLIError, "Unsupported format: '#{self[:format]}'"
       end
+
+      self[:transports].each_value do |v|
+        timeout_value = v[:connect_timeout]
+        unless timeout_value.to_s =~ /^\d*$/ || timeout_value.nil?
```

Comment:
If the config file is YAML (with typed data) and the CLI parser checks types, do we need to allow strings of digits here, rather than just enforcing `timeout_value.is_a?(Integer)`?

Reply:
I wasn't aware that the CLI parser was checking types.  If thats the case then yeah, `timeout_value.is_a?(Integer)` seems better.

Reply:
Yep, [here's an example](https://github.com/puppetlabs/bolt/blob/master/lib/bolt/cli.rb#L151).

Reply:
Updated